source/language/types.rst

```diff
@@ -279,25 +279,53 @@ their argument expression in parentheses:
 
    .. table:: Built-in mathematical functions in OpenQASM3.
 
-      +--------+
-      | arccos |
-      +--------+
-      | arcsin |
-      +--------+
-      | arctan |
-      +--------+
-      | cos    |
-      +--------+
-      | exp    |
-      +--------+
-      | ln     |
-      +--------+
-      | sin    |
-      +--------+
-      | sqrt   |
-      +--------+
-      | tan    |
-      +--------+
+      +----------+--------------------------------+---------------------+
+      | Function | [Input Range/Type] , [...]     | [Output Range/Type] |
+      +==========+================================+=====================+
+      | arccos   | [-1, 1]                        | [0, pi rad]         |
+      +----------+--------------------------------+---------------------+
+      | arcsin   | [-1, 1]                        | [-pi/2, pi/2 rad]   |
+      +----------+--------------------------------+---------------------+
+      | arctan   | [-∞, Real, ∞]                  | [-pi/2, pi/2 rad]   |
+      +----------+--------------------------------+---------------------+
+      | ceiling  | [float], [float]               | [float]             |
+      +----------+--------------------------------+---------------------+
+      | cos      | [float]                        | [float]             |
```

Comment:
Some of this is slightly confusing to me - sometimes we use ranges, and sometimes we use types.  Would it be appropriate to give the types for everything, and then give further limitations on ranges where applicable?  What do we think a sensible response would be when the function is called with something out-of-range?  Should we leave it implementation-defined (hardware interrupt?) or we could specify that (for `float` at least) the functions return `NaN`?  Being low-level, we don't have an exception mechanism, nor any real error handling.

Reply:
Sounds reasonable. I'm moving my household and data setup between the holidays so I am counting on being back in action in the New Year in time for the Pragma WG meeting but not sure before.  I'll look at all these and do what I can when I can. Thanks for your attention to this PR, @jakelishman and happy holidays!

Reply:
The missing ranges and missing types will have to be specified. Maybe we can edit this together online sometime.